content/docs/command-reference/stage/list.md

```diff
@@ -0,0 +1,78 @@
+# stage list
+
+List <abbr>stages</abbr> in the `dvc.yaml`.
+
+## Synopsis
+
+```usage
+usage: dvc stage list [-h] [-q | -v]
+                      [-R] [--all] [--fail]
+                      [--names-only]
+                      [targets ...]
+
+positional arguments:
+  targets          Show stages from a dvc.yaml file or a directory (with -R).
+                   'dvc.yaml' by default.
+```
+
+## Description
+
+`dvc stage list` is useful for quick listing of stages present in the dvc.yaml
+file without having to open it.
+
+It provides a list of stages, with a stage name and the description of the
+stage. The description of the stage is read from
+[`desc` field in the `dvc.yaml` file](doc/user-guide/project-structure/pipelines-files#stage-entries),
+or if it does not exist, DVC generates the short description from the other
+metadata it has about the stage.
+
+DVC also uses this command to provide autocompletion in `bash` and `zsh`.
+
+## Options
+
+- `targets` (optional command argument) - Stages to list from (`./dvc.yaml` by
+  default). Different things can be provided as targets depending on the flags
+  used (more details in each option). Examples:
+
+  - `dvc stage list linear/dvc.yaml`: Specific `dvc.yaml` file to reproduce
+  - `dvc stage list -R pipelines/`: Directory path to explore recursively for
+    `dvc.yaml` files
+  - `dvc stage list train-model`: Specific stage name or a
+    [foreach group](doc/user-guide/project-structure/pipelines-files#foreach-stages)
+    in `./dvc.yaml`
+  - `dvc stage list modeling/dvc.yaml:prepare`: Stage or a
+    [foreach group](doc/user-guide/project-structure/pipelines-files#foreach-stages)
+    in a specific `dvc.yaml` file
+
+- `-R`, `--recursive` - looks for `dvc.yaml` files to list from in any
+  directories given as `targets`, and in their subdirectories. If there are no
+  directories among the targets, this option has no effect.
+
+- `--all` - lists all of the stages present in the repo, regardless of the
```

Comment:
does it looks for all `dvc.yaml`s? may be we can be a bit more explicit on this (it is alternative `dvc stage list -R .`?)

Reply:
Yes, it looks for all the `dvc.yaml` files inside the repo, but it's not the same as `dvc stage list -R .` but to the `-R $(dvc root)`.

Reply:
@shcheklein, something like this?

```suggestion
- `--all` - lists all of the stages from all `dvc.yaml` files in the repo, regardless of the
```

Reply:
sounds good!